src/cargo/util/toml/mod.rs

```diff
@@ -979,6 +979,22 @@ impl StringOrVec {
     }
 }
 
+#[derive(Clone, Debug, Deserialize, Serialize, Eq, PartialEq)]
+#[serde(untagged, expecting = "expected a string or an integer")]
+pub enum StringOrI64 {
+    String(String),
+    I64(i64),
```

Comment:
Why an `i64` instead of a `u64`?

Reply:
I have tried an u32 at the start and it didn't work, as in it compiled but the tests as added by this PR failed. The problem was I think that serde represents toml integers via i64 in its data model. In the end I don't think it matters that much: a negative number will still give an error one way or another.